internal/blocksync/pool.go

```diff
@@ -130,23 +130,22 @@ func (*BlockPool) OnStop() {}
 
 // spawns requesters as needed
 func (pool *BlockPool) makeRequestersRoutine(ctx context.Context) {
-	for {
-		if !pool.IsRunning() {
-			break
-		}
+	timer := time.NewTimer(0)
+	<-timer.C
```

Comment:
I don't think you need/want this line.

Reply:
From an experiment here it seems like it is necessary: https://go.dev/play/p/Trkq1pnyFuH - not draining the channel (i.e. by leaving the line commented out) results in nondeterministic behavior when running that example. Occasionally the reset works, but the majority of the time the program prints `Timed out after 0s`. If you uncomment the draining of the timer channel, it works consistently as expected (`Timed out after 5s`).

As per https://pkg.go.dev/time#Timer.Reset